src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1214,88 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
+                             */
+                            string getJobIdsFromDatabaseQueryString = 
+                                "DECLARE @script as varchar(MAX)\n" +
+                                "SET @script =\n" + 
+                                "'USE [?];\n" +
+                                "IF EXISTS\n" + 
+                                "(SELECT * FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = N''nb_template'')\n"+
+                                "BEGIN\n" + 
+                                "select dbo.nb_template.*,\n" +
+                                "DB_NAME() as db_name,\n" +
+                                "msdb.dbo.sysjobs.*\n" +
+                                "from [?].dbo.nb_template\n" +
+                                "LEFT JOIN\n" + 
+                                "msdb.dbo.sysjobs \n" +
+                                "ON \n" + 
+                                "[?].dbo.nb_template.job_id = msdb.dbo.sysjobs.job_id\n" +
+                                "END '\n" +  
+                                "EXEC sp_MSforeachdb @script\n";
+                            var agentnotebooks = new List<NotebookJobInfo>();
+                            using( SqlCommand getJobIdsFromDatabaseQueryCommand = new SqlCommand(getJobIdsFromDatabaseQueryString, connection))
+                            {
+                                SqlDataAdapter jobIdsAdapter = new SqlDataAdapter(getJobIdsFromDatabaseQueryCommand);
+                                DataSet jobIdsDataSet = new DataSet();
+                                jobIdsAdapter.Fill(jobIdsDataSet);
+                                var serverConnection = ConnectionService.OpenServerConnection(connInfo);
+                                var fetcher = new JobFetcher(serverConnection);
+                                var filter = new JobActivityFilter();
+                                // Fetching all the jobs
+                                var jobs = fetcher.FetchJobs(filter);
+                                // Creating Dictionary of jobs for effcient access later on.
+                                Dictionary<Guid, JobProperties> allJobsHashTable = new Dictionary<Guid, JobProperties>();
+                                if(jobs != null)
+                                {
+                                    foreach (var job in jobs.Values)
+                                    {
+                                        allJobsHashTable.Add(job.JobID, job);
+                                    }
+                                }                                
+                                foreach (DataTable templateTable in jobIdsDataSet.Tables)
+                                {
+                                    foreach (DataRow templateRow in templateTable.Rows){
+                                        NotebookJobInfo notebookJob = new NotebookJobInfo();
+                                        notebookJob.Template = (string)templateRow["json"];
```

Comment:
will this throw if the "json" value is `null`?  perhaps using `as` to cast is safer.